src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -169,20 +170,41 @@ public ConditionalPublisherDescriptor getDescriptor() {
     }
 
     public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.prebuild(condition, publisher, build, listener)) {
-                return false;
-            }
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.prebuild(condition, new RunAllBuilder(getPublisherList()), build, listener);
         }
-        return true;
+        return runner.prebuild(condition, getPublisherList().get(0), build, listener);
     }
 
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.perform(condition, new RunAllBuilder(getPublisherList()), build, launcher, listener);
```

Comment:
[JENKINS-27171](https://issues.jenkins-ci.org/browse/JENKINS-27171)
Now actions in a condition will work like this:
```
if(condition) {
    action1;
    action2;
}
```

was:
```
if(condition) action1;
if(condition) action2;
```
This resolves the problem pointed by @KostyaSha : If runner return false, then next runner/actions wouldn't run at all. This is good for performance for not doing some console string search every time.

If a user really need the condition evaluated each time, they can define multiple conditional actions.